### PR TITLE
feat(connectors): fixes MGDCTRS-717, save agent serviceaccount secret with connector cluster

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description
Saves serviceaccount client secret in connector_cluster on cluster creation to avoid looking it up when providing addon parameters.

## Verification Steps
Cluster creation tests in integration tests MUST continue to work as before. 

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
- [X] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
~~- [ ] Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
~~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~~
~~- [ ] Required Standard Operating Procedure (SOP) is added.~~
~~- [ ] JIRA has created for changes required on the client side~~

### CODE REVIEW

**Comment by dhirajsb** [comment]
@valdar please review and let me know what you think.

File: internal/connector/internal/handlers/connector_cluster.go [hunk 1]
```diff
@@ -244,22 +245,22 @@ func (o *ConnectorClusterHandler) buildAddonParams(serviceAccount *api.ServiceAc
...
-			Value: serviceAccount.ClientSecret,
+			Value: cluster.ClientSecret,
 		},
 	}
 	return p
 }
 
-func (o *ConnectorClusterHandler) buildTokenURL(serviceAccount *api.ServiceAccount) (string, error) {
+func (o *ConnectorClusterHandler) buildTokenURL(cluster dbapi.ConnectorCluster) (string, error) {
 	u, err := url.Parse(o.KeycloakConfig.KafkaRealm.TokenEndpointURI)
```
Comment (akoserwal) [inline internal/connector/internal/handlers/connector_cluster.go:259]: "KeycloakConfig.KafkaRealm.TokenEndpointURI" this needs to be set on basis of which sso provider is used. `https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/keycloak/config.go#L89`

if the provider is MAS_SSO use the KafkaRealm.
If the provider is REDHAT_SSO use the RedhatSSORealm

https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/services/sso/keycloak.go#L26 will return the set sso service/provider realm

Comment (akoserwal) [inline internal/connector/internal/handlers/connector_cluster.go:259]: `KeycloakConfig.KafkaRealm.TokenEndpointURI`should be set on the basis of SelectSSOProvider
* if MAS_SSO then `KeycloakConfig.KafkaRealm.TokenEndpointURI`
* if REDHAT_SSO then `KeycloakConfig.RedhatSSORealm.TokenEndpointUR`

or use this which will return the set sso provider
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/services/sso/keycloak.go#L26

Comment (akoserwal) [reply internal/connector/internal/handlers/connector_cluster.go:259]: Something like: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/894

**Review by valdar — CHANGES_REQUESTED** [review]
add issue for missing tests and address @akoserwal comments, other than that it lgtm.

**Comment by dhirajsb** [comment]
There is already an integration test for addon parameters. The agent APIs in `connector-agent-api.feature` won't work without being able to retrieve the client id and secret from the new column in the DB.

**Comment by dhirajsb** [comment]
@lburgazzoli ideally we should also rename the addon-parameters for base url and realm:
```
mas-sso-base-url - > sso-base-url
mas-sso-realm - > sso-realm
```
I'm updating this PR to use the sso service agnostic values. We can update these param names later to resolve https://issues.redhat.com/browse/MGDCTRS-721

**Comment by dhirajsb** [comment]
@valdar I updated the new `client_secret` column to have a default value of '' to allow existing connector clusters in `dev` to continue working. Their addon parameters have already been generated, so no need for restoring the client secret in the DB. The clusters in `staging` are going to be deleted for a previous multi-tenancy migration so this won't be an issue.

**Review by valdar — APPROVED** [review]

**Pull request merged**